Enforce rustfmt on CI

pyfisch opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/22126)
<!-- Reviewable:end -->

pyfisch committed:
Run rustfmt on test-tidy

Add ./mach fmt command.
Mach installs rustfmt if needed.

pyfisch committed:
Format remaining files

pyfisch committed:
Manually format remaining problems

Use line comments instead of block comments
as block comments contain trailing whitespace
after formatting with rustfmt.

Skip tests for malloc_size_of and script_plugins
with rustfmt as they have many block comments.

highfive commented:
Heads up! This PR modifies the following files:
 * @jgraham: components/webdriver_server/lib.rs
 * @wafflespeanut: python/servo/devenv_commands.py, python/servo/testing_commands.py
 * @emilio: components/style/gecko_bindings/sugar/ns_css_value.rs, components/style/dom_apis.rs, components/style/gecko/url.rs, components/style/animation.rs, components/style/values/specified/text.rs and 58 more
 * @kichjang: components/script/dom/request.rs, components/script/dom/timeranges.rs, components/script/dom/htmlcanvaselement.rs, components/script/textinput.rs, components/script/dom/eventtarget.rs and 80 more
 * @asajeffrey: components/script/dom/request.rs, components/script/dom/timeranges.rs, components/script/dom/htmlcanvaselement.rs, components/script/textinput.rs, components/script/dom/eventtarget.rs and 77 more
 * @cbrewster: components/constellation/network_listener.rs, components/constellation/constellation.rs, components/constellation/pipeline.rs
 * @paulrouget: components/compositing/build.rs, components/constellation/network_listener.rs, components/constellation/constellation.rs, components/servo/lib.rs, components/constellation/pipeline.rs and 1 more

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

emilio commented:
If we do this, which makes more annoying for me to port Gecko changes, can we enable the reordering of imports and such, and remove the tidy one?

At least that'd be roughly neutral or even positive on the effort it takes me :)

Also, maybe, remove the `match_block_trailing_comma` thing while at it?

emilio commented on python/servo/testing_commands.py:
```diff
@@ -304,6 +304,12 @@ def test_content(self):
 
+    def install_rustfmt(self):
+        if self.call_rustup_run(["cargo", "fmt", "--version", "-q"],
+                                stderr=open(os.devnull, "w")) != 0:
```
Doesn't this leak the file descriptor?

pyfisch replied:
Good catch!

jdm commented:
Let's have two more commits - one that disables the tidy check for import ordering, and one that enables rustfmt import reordering and formats the whole tree with it.

pyfisch committed:
Remove import order check from test-tidy

Fix leaking file descriptor.

pyfisch committed:
Reorder imports

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 5de5fab has been approved by `jdm`

<!-- @bors-servo r=jdm 5de5fab6c8567884fc420a79c5c32227f2b59124 -->

jdm commented:
@bors-servo r-
test-tidy complains about some line lengths now.

jdm commented:
Also apparently travis is not set up to run cargo fmt yet:
```
./components/script/dom/bluetoothcharacteristicproperties.rs:6: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattcharacteristic.rs:8: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattcharacteristic.rs:10: Line is longer than 120 characters
./components/script/dom/bluetoothremotegattdescriptor.rs:8: Line is longer than 120 characters
Running the dependency licensing lint...
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
```

jdm commented:
And the selftests failed:
```
======================================================================
FAIL: test_rust (servo_tidy_tests.test_tidy.CheckTidiness)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/servo/servo/python/tidy/servo_tidy_tests/test_tidy.py", line 108, in test_rust
    self.assertEqual('extra space after {', errors.next()[2])
AssertionError: 'extra space after {' != 'missing space before }'
----------------------------------------------------------------------
Ran 28 tests in 0.030s
FAILED (failures=1)
The command "./mach test-tidy --no-progress --self-test" exited with 1.
```

pyfisch commented:
I've pushed the two commits as suggested by @jdm.

There are still a few problems with this PR:

* On Travis CI the first build does not install rustup. For this reason `./mach test-tidy` fails now. Install rustup there too?
* I used `#![rustup::skip]`  in two files to stop rustfmt from mangling whitespace (introduce trailing whitespace to block comments) but this causes the error `error[E0658]: non-builtin inner attributes are unstable`. What to do here?
* Rustfmt formats imports in 4 cases such that the lines are longer than 120 characters which causes a tidy error: `use dom::bindings::codegen::Bindings::BluetoothCharacteristicPropertiesBinding::BluetoothCharacteristicPropertiesMethods;`
* > Also, maybe, remove the match_block_trailing_comma thing while at it?
    I think this a good idea. It does not disrupt test-tidy.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #22086) made this pull request unmergeable. Please resolve the merge conflicts.

pyfisch committed:
Fix test-tidy self test

Install rustup on first Travis job.
Only use rustfmt::skip as an outer attribute.

pyfisch commented:
Rebased and fixed. Lets see if it works in Travis.

jdm commented:
```
tidy reported no errors.
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment
The command "./mach test-tidy --no-progress --all" exited with 1.
```

pyfisch commented:
Travis now passes.

AppVeyor has some connectivity issues:
>  The remote server returned an error: (503) Server Unavailable. Service Unavailable

bors-servo commented:
:hourglass: Testing commit 1855c88d6106c73febec1f5f29e246a75ff2f849 with merge da2d9b2228441863feba2af2a60f84d2657fb962...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/13055), [android-mac](http://build.servo.org/builders/android-mac/builds/374), [android-x86](http://build.servo.org/builders/android-x86/builds/1177), [arm32](http://build.servo.org/builders/arm32/builds/13224), [arm64](http://build.servo.org/builders/arm64/builds/13258), [linux-dev](http://build.servo.org/builders/linux-dev/builds/13351), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/10434), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10471), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/13325), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/5805), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/5810), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/10082), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10125), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/5892), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/5818), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/cNDsoZ5gRRCXnOZZDizEAQ), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/8714)
Approved by: jdm
Pushing da2d9b2228441863feba2af2a60f84d2657fb962 to master...

SimonSapin commented:
Nice! @pyfisch, are you subscribed to the [dev-servo](https://lists.mozilla.org/listinfo/dev-servo) mailing list? I think it’d be nice to send an announcement there with:

* A link to this PR
* Format checking is now part of `./mach test-tidy`, so unformatted code will fail CI
* `./mach fmt` does the formatting
* Import ordering has changed (but `mach fmt` will fix it)

If you prefer I can do it for you :)

pyfisch commented:
Sent an email.
